### PR TITLE
add lighthouse benchmark in Github Actions

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

New Job is added to test.yml workflow which takes in the dynamic docs url based on the branch name and run lighthouse benchmark on index.html
For now only the index page is benchmarks but we will eventually add all the routes so that performance profiling runs on every component routes.

Lighthouse config

Only Performance is being audited now with dom-size audit settings. We can expand this based on our need.

## Motivation and context

Lighthouse reporting is essential to keep our docs page faster and also to run performance profiling on SWC components.

## How has this been tested?

-   [x] _Test case 1_
https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1707302671447-43007.report.html

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by Westbrook — COMMENTED** [review]
A version of this that added `.lighthouseci` to the `.gitignore` and uses the config to allow our current usage to "pass" would be a great first step here.

With the tool in repo, we could then create issues/branches for each test that would normally fail and use those to incrementally get back to the performance we'd want to target here.

Once `index.html` was passing, we could then move to other pages in the site, too.

File: packages/radio/src/radio.css [hunk 1]
```diff
@@ -23,7 +23,7 @@ governing permissions and limitations under the License.
 /**
  * Work around for https://github.com/adobe/spectrum-css/issues/2251
  **/
-:host([dir="rtl"]) #button:after {
+:host([dir='rtl']) #button:after {
```
Comment (castastrophe) [inline packages/radio/src/radio.css:26]: This seems unrelated to the rest of the PR - was this intentionally included?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    lighthouseci:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+            - uses: actions/setup-node@v3
+              with:
+                  node-version: '18'
+                  cache: 'yarn'
+                  registry-url: 'https://registry.npmjs.org'
+            - run: npm install -g @lhci/cli@0.12.x
```
Comment (castastrophe) [inline .github/workflows/main.yml:13]: Since lighthouse is installed as part of the monorepo, you shouldn't need to install it separately here. The yarn install that follows will include the lighthouse cli.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            - name: Install dependencies
+              run: yarn --frozen-lockfile
+            - run: yarn docs:build
+            - run: lhci autorun
```
Comment (castastrophe) [inline .github/workflows/main.yml:17]: Instead of running this here, I would use the yarn command `yarn lighthouse` so that you're using the locally installed version.

File: lighthouserc.cjs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+module.exports = {
+    ci: {
+        collect: {
+            url: [
+                'http://localhost:8000/',
```
Comment (castastrophe) [inline lighthouserc.cjs:16]: Do the docs always use 8000 or does it fallback to an open port if that is in use? I ask because you might need to use an environment variable or something else here.

**Review by castastrophe — CHANGES_REQUESTED** [review]

File: lighthouserc.cjs (hunk 4 above)
Comment (Rajdeepc) [reply lighthouserc.cjs:16]: Docs will always use 8000 as it is 11ty default port

File: .github/lighthouse/lighthouserc.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "ci": {
+        "collect": {
+            "settings": {
+                "onlyCategories": ["performance"],
```
Comment (Westbrook) [inline .github/lighthouse/lighthouserc.json:5]:
```suggestion
                "onlyCategories": [
                    // Excluded categories do not currently pass
                    // "accessibility",
                    // "best-practices",
                    "performance",
                     // "pwa",
                     // "seo"
                ],
```
A

File: .github/lighthouse/lighthouserc.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+                "onlyAudits": ["dom-size"]
```
Comment (Westbrook) [inline .github/lighthouse/lighthouserc.json:6]: Is there an equivalent list we could include like above?

**Review by Westbrook — COMMENTED** [review]
Getting pretty close! Just want to document some next steps in line of the code and we should be good to go.

File: .github/lighthouse/lighthouserc.json (hunk 5 above)
Comment (Rajdeepc) [reply .github/lighthouse/lighthouserc.json:5]: I don't think we can add comments in json. :(

Comment (Westbrook) [reply .github/lighthouse/lighthouserc.json:5]: Can we use a JS config like this one? https://github.com/GoogleChrome/lighthouse/blob/main/docs/configuration.md#usage

Comment (Rajdeepc) [reply .github/lighthouse/lighthouserc.json:5]: Updated to
```
export default {
    extends: 'lighthouse:default',
    settings: {
        onlyCategories: [
            // Excluded categories do not currently pass
            // "accessibility",
            // "best-practices",
            'performance',
            // "pwa",
            // "seo"
        ],
        onlyAudits: ['first-meaningful-paint', 'speed-index', 'interactive'],
    },
};
```

File: .github/lighthouse/lighthouserc.json (hunk 6 above)
Comment (Rajdeepc) [reply .github/lighthouse/lighthouserc.json]: Yes as here
https://github.com/GoogleChrome/lighthouse/blob/ed04b10ccfa4e82ee5cd5b437e474a12c1ba52d0/core/config/default-config.js#L178

Comment (Rajdeepc) [reply .github/lighthouse/lighthouserc.json]: also added in the config
